pkg/deploy/generator/resources.go

```diff
@@ -1089,6 +1052,124 @@ EOF
 
 chcon -R system_u:object_r:var_log_t:s0 /var/opt/microsoft/linuxmonagent
 
+cat >/etc/systemd/system/download-mdm-credentials.service <<'EOF'
```

Comment:
Please wrap this with `for <var> in mdm mdsd; do ... done` so that the systemd file contents are not duplicated twice in this script.